configuration/src/agent/signer.rs

```diff
@@ -1,16 +1,14 @@
+use std::str::FromStr;
+
 use crate::FromEnv;
 use nomad_types::HexString;
-use serde_json::json;
 
 /// Ethereum signer types
 #[derive(Debug, Clone, PartialEq, serde::Deserialize)]
-#[serde(tag = "type", rename_all = "camelCase")]
+#[serde(untagged, rename_all = "camelCase")]
 pub enum SignerConf {
-    /// A local hex key
-    HexKey {
-        /// Hex string of private key, without 0x prefix
-        key: HexString<64>,
-    },
+    /// A local hex key, hex string of private key, with or without 0x prefix
+    HexKey(HexString<64>),
```

Comment:
The env variable names no longer trace the JSON structure anymore. 

```
"transactionSigners": {
   "ethereum": "0x1111111111111111111111111111111111111111111111111111111111111112",
}

vs.

TRANSACTIONSIGNERS_ETHEREUM_KEY=0x1111111111111111111111111111111111111111111111111111111111111112
```

Would prefer to leave the as `HexKey { key: HexString<64> }` for consistency between JSON and env. So we'd have:
```
"transactionSigners": {
   "ethereum": {
      "key": "0x1111111111111111111111111111111111111111111111111111111111111112",
   }
}
```

Reply:
I'm not sure that this is desirable. Generally, an object with only 1 key is an anti-pattern. I think a lesson of the previous config management system was that the env var convention should be based on what's easiest for deployment, rather than the json schema. I chose to not change the env handling in order to avoid requiring any changes in env var inject

Reply:
Ok in place of this then, we'll need clearer docs about env variable naming (we can even just use env.test as the example/docs). We can also remove most of the crazy names then too. Lets merge this, then eventually make [another PR for simplifying env vars](https://github.com/nomad-xyz/rust/issues/148)